src/vm/dllimport.cpp

```diff
@@ -7047,7 +7055,6 @@ HINSTANCE NDirect::LoadLibraryModule(NDirectMethodDesc * pMD, LoadLibErrorTracke
     bool libNameIsRelativePath = Path::IsRelative(wszLibName);
     if (hmod == NULL)
     {
-#ifndef FEATURE_CORECLR
         // First checks if the method has DefaultDllImportSearchPathsAttribute. If method has the attribute
```

Comment:
This seems broken before your change -- `searchAssemblyDirectory` is by default set to true, so the copy next to the assembly will always be picked up over host specified paths. If the host specified a zero-day patched assembly location, that will never get used -- @gkhanna79.

Can this be set to FALSE instead or the load be done after the host specified directories?

> DefaultDllImportSearchPath would still get overriden by host native dll search path settings
Once we do the above, then we should make sure the host specified paths are not overridden by `DefaultDllImportSearchPath` 's setting of the `searchAssemblyDirectory` flag.